pkg/k8s/apis/cilium.io/v2alpha1/bgp_advert_types.go

```diff
@@ -92,13 +121,22 @@ type Advertisement struct {
 	Attributes *CiliumBGPAttributes `json:"attributes,omitempty"`
 }
 
+// ServiceAddressOptions defines the configuration for Service advertisement type.
+type ServiceAddressOptions struct {
```

Comment:
Maybe just `ServiceOptions`, if we later needed to extend it with some options not related to addresses?

Also, should the exported types have some common prefix, like `CiliumBGP` o `BGP`? We seem to be a bit inconsistent about naming of public/exported types in this file.

Reply:
yep makes sense. I will take this opportunity to fix the export types.

Reply:
I went with prefix of BGP, since that seems more natural.

Reply:
Ah, missed few - will fix them now